samples/test-samples.cmake

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script requires the variables SOURCE_DIR, BUILD_DIR, and
+# PREFIX_DIR to be defined:
+#
+#     cmake -DSOURCE_DIR=~/openenclave -DBUILD_DIR=~/openenclave/build -DPREFIX_DIR=/opt/openenclave -P ~/openenclave/samples/test-samples.cmake
```

Comment:
The script doesn't assert that these variables are defined though? It seems like it would allow you to make to the current folder by default?

Reply:
This script isn't meant to be used outside of our test; I don't see a reason to add a nice UI to it.

Reply:
It's fine, but the observation is less about nice UI and more about defensive coding; it's parameterized rather than hardcoded i.e. there's an assumption of flexible use beyond it's invocation from samples/CMakeLists.txt, otherwise, it would be more direct to assign `${CMAKE_CURRENT_SOURCE_DIR}` internal to this script and less prone to usage error.

Reply:
It can't be hard-coded because the input directories change based on the location of the build (which includes absolute directories, so it changes with the environment). The upstream build (that is, Open Enclave) has this knowledge, but the downstream build doesn't have access to it, and they can't be relative paths (since you could put a build folder anywhere; in fact, I frequently nest them). I admit it's hacky, but we have to pass that info to it somehow. I wouldn't say the script is flexible in use beyond this scenario (at least, not without more work).